hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -89,4 +98,12 @@ private Message process(final SCMRatisRequest request)
     }
   }
 
+  @Override
+  public long takeSnapshot() throws IOException {
+    LOG.info("Current Snapshot Index {}", getLastAppliedTermIndex());
+    TermIndex lastTermIndex = getLastAppliedTermIndex();
```

Comment:
I believe we don't need to. 

```
    public TermIndex getLastAppliedTermIndex() {
        return (TermIndex)this.lastAppliedTermIndex.get();
    }
```

```
    protected boolean updateLastAppliedTermIndex(long term, long index) {
        TermIndex newTI = TermIndex.newTermIndex(term, index);
        TermIndex oldTI = (TermIndex)this.lastAppliedTermIndex.getAndSet(newTI);
         ....
```

Both function access `this.lastAppliedTermIndex`.

Reply:
Since all rocksdb writes are batched to write to disk in takesnapshot,  we need to change takesnapshot default frequency of ratis during scm ratis server creation.

Reply:
> Since all rocksdb writes are batched to write to disk in takesnapshot, we need to change takesnapshot default frequency of ratis during scm ratis server creation.

https://issues.apache.org/jira/browse/HDDS-4636 I will prefer to handle separately.